raiden/blockchain/state.py

```diff
@@ -21,9 +20,14 @@ def get_channel_state(
         reveal_timeout: BlockTimeout,
         payment_channel_proxy,
         opened_block_number: BlockNumber,
-        opened_block_hash: BlockHash,
 ):
-    channel_details = payment_channel_proxy.detail(opened_block_hash)
+    # Here we have to query the latest state as during restart a lot of things
+    # can have happened, such as the channel having been closed. To get this
+    # information we should use the latest block hash
+    latest_block_hash = payment_channel_proxy.client.blockhash_from_blocknumber(
+        block_number=payment_channel_proxy.client.block_number(),
```

Comment:
this is doing two rpc calls.

Reply:
We use that a lot in other places. I can change that elsewhere too. I suppose you mean do a `getBlock('latest')` and retrieve the hash?

Reply:
Ah no. as simple as `payment_channel_proxy.client.blockhash_from_blocknumber('latest')`